kernel/include/kernel_internal.h

```diff
@@ -202,6 +202,11 @@ void z_mem_manage_init(void);
 /* Workaround for build-time page table mapping of the kernel */
 void z_kernel_map_fixup(void);
 
+#define LOCKED(lck) for (k_spinlock_key_t __i = {},			\
```

Comment:
A comment/warning on the safe use of the macro would be nice here: ensure there is no break/goto out of the body of the locked section to avoid a dangling spinlock.